src/main/java/org/protege/editor/owl/server/http/handlers/HTTPChangeService.java

```diff
@@ -56,40 +59,80 @@ public void handleRequest(HttpServerExchange exchange) {
 
 	private void handlingRequest(HttpServerExchange exchange)
 			throws IOException, ClassNotFoundException, LoginTimeoutException, ServerException {
+		ObjectInputStream ois = new ObjectInputStream(exchange.getInputStream());
+		ProjectId projectId = (ProjectId) ois.readObject();
```

Comment:
Since this is user input, I think you should probably verify it. I.e. check that projectId isn't null here.